app/assets/javascripts/lara-api.js

```diff
@@ -140,15 +143,33 @@ window.LARA = {
       remove: remove
     }
   },
+  /****************************************************************************
+  @deprecated saveLearnerState
+  @see saveLearnerPluginState
+  ****************************************************************************/
+  saveLearnerState: function (pluginId, state) {
+    var deprication =
+      '⚠️ saveLearnerState is depricated, use `saveLearnerPluginState` instead'
+    console.warn(deprication);
+    this.saveLearnerPluginState(pluginId, state);
+  },
 
   /****************************************************************************
-   @function saveUserState: Ask LARA to save the users state for the plugin
-   @arg {ILaraPluginRef} pluginInstance - The plugin trying to save data
+   @function saveLearnerPluginState: Ask LARA to save the users state for the plugin
+   @arg {string} pluginId - ID of the plugin trying to save data, initially passed to plugin constructor in the context
    @arg {string} state - A JSON string representing serialized plugin state.
+   @example
+    LARA.saveLearnerPluginState(plugin, '{"one": 1}').then((data) => console.log(data))
    @returns Promise
   ****************************************************************************/
-  saveUserState: function (pluginInstance, state) {
-    console.log('Plugin', pluginInstance, 'wants to save a state:', state);
+  saveLearnerPluginState: function (pluginId, state) {
+    var pluginsApi = window.Plugins;
+    if (pluginsApi && typeof pluginsApi.saveLearnerPluginState === 'function') {
+      return pluginsApi.saveLearnerPluginState(pluginId, state);
+    }
+    return new Promise( function(resolve, reject) {
+      reject('pluginApi not defined. saveLearnerPluginState failed.')
+    });
```

Comment:
Is there any scenario when this can happen (PluginApi not available)? I guess it would be some total collapse of LARA JS env. Actually, it might be even beneficial if everything fails, some errors are thrown, don't get handled, and finally Rollbar reports them to us. 

In general, I'm not sure if the nice handling of unexpected disasters is desired. It might silence any signs that they happened. Plugin might not care about this Promise and the fact it got rejected. I'm not too concerned about it, as I think it's unlikely this line of code will be ever executed tho. Just an observation. :wink: